cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -174,6 +174,56 @@ def test_verify_revert_snapshot(full_test_vm, provider, soft_assert, register_ev
 
 
 @pytest.mark.uncollectif(lambda provider: not provider.one_of(VMwareProvider))
+def test_operations_suspended_vm(small_test_vm, provider, soft_assert):
+    """Tests snapshot operations on suspended vm
+
+    Metadata:
+        test_flag: snapshot, provision
+    """
+    # Create first snapshot when VM is running
+    snapshot1 = new_snapshot(small_test_vm)
+    snapshot1.create()
+    wait_for(lambda: snapshot1.active, num_sec=300, delay=20, fail_func=snapshot1.refresh)
+    # Suspend the VM
+    small_test_vm.power_control_from_cfme(option=small_test_vm.SUSPEND, cancel=False)
+    small_test_vm.wait_for_vm_state_change(desired_state=small_test_vm.STATE_SUSPENDED)
+    current_state = small_test_vm.find_quadicon().state
+    soft_assert(current_state.startswith('currentstate-suspended'),
+                "Quadicon state is {}".format(current_state))
+    # Create second snapshot when VM is suspended
+    snapshot2 = new_snapshot(small_test_vm)
+    snapshot2.create()
+    wait_for(lambda: snapshot2.active, num_sec=300, delay=20, fail_func=snapshot2.refresh)
+    # Try to revert to first snapshot while the VM is suspended
+    snapshot1.revert_to()
+    wait_for(lambda: snapshot1.active, num_sec=300, delay=20, fail_func=snapshot1.refresh)
+    # Check VM state, VM should be off
+    current_state = small_test_vm.find_quadicon().state
+    soft_assert(current_state.startswith('currentstate-off'),
+                "Quadicon state is {}".format(current_state))
+    assert small_test_vm.provider.mgmt.is_vm_stopped(small_test_vm.name)
+    # Revert back to second snapshot
+    snapshot2.revert_to()
+    wait_for(lambda: snapshot2.active, num_sec=300, delay=20, fail_func=snapshot2.refresh)
+    # Check VM state, VM should be suspended
+    current_state = small_test_vm.find_quadicon().state
+    soft_assert(current_state.startswith('currentstate-suspended'),
+                "Quadicon state is {}".format(current_state))
+    assert small_test_vm.provider.mgmt.is_vm_suspended(small_test_vm.name)
+    # Try to delete both snapshots while the VM is suspended
+    snapshot2.delete_all()
+    # Wait for both snapshots to disappear
+    try:
+        wait_for(lambda: not snapshot1.exists, num_sec=300, delay=20, fail_func=snapshot1.refresh)
+    except NameError:
```

Comment:
@apagac  I don't understand why `NameError` is caught here? Should it be `TimeOutError`?

Reply:
So this wait_for is used as a check if the snapshot is really deleted from the VM. If all goes well, the snapshot we are looking for is not there, and snapshot1.exists throws NameError exception.

Reply:
Snapshot.exists shouldn't throw NameError exception, it should catch that and return `False`, allowing for much better flow here.

I'd like to scope that change to this PR to avoid implementing this flow.